File: ios/Modals/TripModal.swift
//
//  TripModal.swift
//  Glimpse
//
//  Created by Nitin Issac Joy on 10/07/19.
//  Copyright © 2019 Facebook. All rights reserved.
//

import Foundation
import RealmSwift

@objcMembers
class TripModal: Object {
  dynamic var profileId: String = ""
  dynamic var tripId: String = ""
  dynamic var tripName: String = ""
  dynamic var countryCode: List<Country> = List<Country>()
  dynamic var temperature : String = ""
  dynamic var daysOfTravel: Int = 0
  dynamic var distanceTravelled : Int = 0
  dynamic var startDate: String = ""
  dynamic var endDate: String = ""
  dynamic var masterImage: String = ""
  dynamic var isPublic: Bool = false
  dynamic var syncComplete: Bool = false
  
  func GetAsDictionary() -> [String:Any] {
    var dict : [String: Any] = [:]
    dict["profileId"] = self.profileId;
    dict["tripId"] = self.tripId;
    dict["tripName"] = self.tripName;
    dict["countryCode"] = self.countryCode;
    dict["temperature"] = self.temperature;
    dict["daysOfTravel"] = self.daysOfTravel;
    dict["distanceTravelled"] = self.distanceTravelled;
    dict["startDate"] = self.startDate;
    dict["endDate"] = self.endDate;
    dict["masterPicURL"] = self.masterImage;
    dict["isPublic"] = self.isPublic;
    dict["syncComplete"] = self.syncComplete;
    
    return dict;
  }
}

File: android/app/src/main/java/com/beerwithai/glimpse/Engine/Modals/TripModal.java
package com.beerwithai.glimpse.Engine.Modals;

import com.facebook.react.bridge.WritableArray;
import com.facebook.react.bridge.WritableMap;
import com.facebook.react.bridge.WritableNativeArray;
import com.facebook.react.bridge.WritableNativeMap;

import java.util.HashMap;
import java.util.Map;

import io.realm.RealmList;
import io.realm.RealmObject;

public class TripModal extends RealmObject {
    public String profileId = "";
    public String tripId = "";
    public String tripName = "";
    public RealmList<String> countryCode = new RealmList<>();
    public String temperature = "";
    public Integer daysOfTravel = 0;
    public Integer distanceTravelled = 0;
    public String startDate = "";
    public String endDate = "";
    public String masterImage = "";
    public boolean isPublic = false;
    public boolean syncComplete  = false;

    public void CopyConstructor(TripModal trip) {
        this.tripName = trip.tripName;
        this.countryCode = trip.countryCode;
        this.temperature = trip.temperature;
        this.daysOfTravel = trip.daysOfTravel;
        this.distanceTravelled = trip.distanceTravelled;
        this.startDate = trip.startDate;
        this.endDate = trip.endDate;
        this.masterImage = trip.masterImage;
        this.isPublic = trip.isPublic;
        this.syncComplete = trip.syncComplete;
    }

    public WritableMap GetAsDictionary() {
        WritableMap obj = new WritableNativeMap();
        obj.putString("profileId", this.profileId);
        obj.putString("tripId", this.tripId);
        obj.putString("tripName", this.tripName);

        WritableArray cc = new WritableNativeArray();
        for(String code: this.countryCode) {
            cc.pushString(code);
        }
        obj.putArray("countryCode", cc);
        obj.putString("temperature", this.temperature);
        obj.putInt("daysOfTravel", this.daysOfTravel);
        obj.putInt("distanceTravelled", this.distanceTravelled);
        obj.putString("startDate", this.startDate);
        obj.putString("endDate", this.endDate);
        obj.putString("masterImage", this.masterImage);
        obj.putBoolean("isPublic", this.isPublic);
        obj.putBoolean("syncComplete", this.syncComplete);
        return obj;
    }
}